feg/gateway/docker/build.py

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env python3
+
+"""
+Copyright 2021 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import argparse
+import json
+import glob
+import os
+import shutil
+import subprocess
+from collections import namedtuple
+from typing import Iterable, List, Optional
+
+
+HOST_MAGMA_ROOT = '../../../.'
+
+def main() -> None:
+    args = _parse_args()
+
+    if args.mount:
+        _run(['up', '-d', 'test'])
+        _run(['exec', 'test', 'bash'])
+        _down(args)
+
+    elif args.lint:
+        _run(['up', '-d', 'test'])
+        _run(['exec', 'test', 'make', 'lint'])
+        _down(args)
+
+    elif args.precommit:
+        _run(['up', '-d', 'test'])
+        _run(['exec', 'test', 'make', 'precommit'])
+        _down(args)
+
+    elif args.coverage:
+        _run(['up', '-d', 'test'])
+        _run(['exec', 'test', 'make', 'cover'])
+        _down(args)
+
+    elif args.tests:
+        _run(['up', '-d', 'test'])
+        _run(['exec', 'test', 'make', 'test'])
+        _down(args)
+
+    elif args.health:
+        _run(['up', '-d'])
+        _run_health()
+        _down(args)
+
+    elif args.git:
+       print(json.dumps(_run_get_git_vars(), indent=4, sort_keys=True))
+
+    else:
+        _run(['build'] + _get_default_build_args(args))
+        _down(args)
+
+def _run(cmd: List[str]) -> None:
+    """ Run the required docker-compose command """
+    cmd = ['docker-compose'] + cmd
+    print("Running '%s'..." % ' '.join(cmd))
+    try:
+        subprocess.run(cmd, check=True)
+    except subprocess.CalledProcessError as err:
+        exit(err.returncode)
+
+def _down(args: argparse.Namespace) -> None:
+    if args.down:
+         _run(['down'])
+
+def _get_default_build_args(args: argparse.Namespace) -> List[str]:
+    ret = []
+    git_info = _run_get_git_vars()
+
+    for arg, val in git_info.items():
+        ret.append("--build-arg")
+        ret.append("{}={}".format(arg, val))
+
+    if args.parallel:
+        ret.append('--parallel')
+    if args.nocache:
+        ret.append('--no-cache')
+    return ret
+
+
+def _run_get_git_vars():
+    try:
+        cmd = "tools/get_version_info.sh"
```

Comment:
is get_version_info.sh missing from this PR?

Reply:
yes! you are right. For some reason when I reseted the commit it was not added 

Now it should be there